src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -998,10 +998,14 @@ public function findDefinition($id)
             $id = (string) $this->aliasDefinitions[$id];
 
             if (isset($seen[$id])) {
-                throw new ServiceCircularReferenceException($id, array_keys($seen));
+                $seen = array_values($seen);
+                $seen = array_slice($seen, array_search($id, $seen));
```

Comment:
the slicing part is not covered by the test (as the tests cover only the case where the loop starts at the beginning of the list).
Can you add another test covering this ?

Reply:
test updated